processor/src/trace.rs

```diff
@@ -21,21 +26,41 @@ impl ExecutionTrace {
             system,
             decoder: _,
             stack,
-            hasher: _,
-            bitwise: _,
-            memory: _,
+            hasher,
+            bitwise,
+            memory,
             advice: _,
         } = process;
 
+        // get the length required to hold all execution trace steps
+        let aux_trace_len = hasher.trace_len() + bitwise.trace_len() + memory.trace_len();
+        let mut trace_len = usize::max(stack.trace_len(), aux_trace_len);
+        // pad the trace length to the next power of 2
+        if !trace_len.is_power_of_two() {
+            trace_len = trace_len.next_power_of_two();
+        }
+
+        // allocate columns for the trace of the auxiliary table
+        let mut aux_table_trace: AuxiliaryTableTrace = (0..AUXILIARY_TABLE_WIDTH)
+            .map(|_| Vec::<Felt>::with_capacity(trace_len))
+            .collect::<Vec<_>>()
+            .try_into()
+            .unwrap();
```

Comment:
nit: would prefer `.expect()` with a message instead of `.unwrap()` in production code.

Reply:
Fixed! Along with a similar instance in the Bitwise module.